packages/integration-tests/src/EntityManager.reactiveRules.test.tsx

```diff
@@ -178,16 +178,41 @@ describe("EntityManager.reactiveRules", () => {
     expect(a.numberOfBooks.get).toBe(0);
   });
 
+  it.withCtx("updates derived fields that are foreign keys", async ({ em }) => {
+    // Given an entity with an async derived field
+    const a = newAuthor(em);
+    const b = newBook(em, { author: a, reviews: [{ rating: 10 }] });
+    await em.flush();
+
+    const favoriteBook = a.favoriteBook.get;
+    // For some reason .toMatchEntity(b) fails here
+    expect(a.favoriteBook.get).toBe(b);
+
+    // If there is a new favorite book
+    const b2 = newBook(em, { author: a, reviews: [{ rating: 20 }] });
+    await em.flush();
+    // Then the derived field is updated
+    expect(a.favoriteBook.get).toBe(b2);
```

Comment:
Suggestion: This is checking the entities internal data and not necessarily what is literally in the database, so maybe add an assertion that calls `select("authors")` and asserts against the `favorite_book_id` column.